Warnings Squashing Round 1

Malkierian opened the pull request:
Mostly fixing implicit conversion warnings, but also a few truncation warnings. 9 files this time. The `AMMO() = CUR_CAPACITY()` lines in `hook_handlers.cpp` still need addressing, but I wasn't sure how the `uint16_t` of `CUR_CAPACITY` was supposed to map to the `int8_t` of `AMMO`, and for some reason, `CUR_CAPACITY` also makes use of an `int16_t` cast internally, so if anyone has any suggestions on that, I'd appreciate it.

<!--- section:artifacts:start -->
### Build Artifacts
  - [soh-linux.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/3118690631.zip)
  - [soh-windows.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/3118696973.zip)
  - [soh-mac.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/3118750841.zip)
<!--- section:artifacts:end -->

Malkierian committed:
Fix truncation warning in `entrance.h`.

Malkierian committed:
Fix type warning in fishsanity.cpp

Malkierian committed:
Fix implicit conversion warnings in BossRush, Mouse, and UIWidgets.hpp.

Malkierian committed:
Add Random_Float and use it in GameInteractor_RawAction.cpp.
Make non-specified seed init a separate function.

Malkierian committed:
clang after type change

Malkierian committed:
Fix truncation and conversion warnings in ExtraTraps.cpp.

Malkierian committed:
Resolve type conversion warnings in InputViewer.cpp.

Malkierian committed:
Resolve some type conversion warnings in hook_handlers.cpp.

Malkierian commented:
Alright, my only concern about this PR has been confirmed to be a non-issue, so this should be ready to go.

serprex commented:
Which compiler are you looking to reduce warnings from?

serprex commented on soh/soh/Enhancements/randomizer/3drando/random.cpp:
```diff
@@ -25,10 +24,22 @@ uint32_t Random(int min, int max) {
 }
 
+// Returns a random float in range [min, max-1]
+float Random_Float(float min, float max) {
+    Random_InitSeed();
+    boost::random::uniform_real_distribution<float> distribution(min, max - 1);
+    return distribution(generator);
+}
+
 // Returns a random floating point number in [0.0, 1.0]
 double RandomDouble() {
```
Bit unintuitive having `Random_Float` & `RandomDouble` be so divergent. I'd lean towards having `RandomFloat` match `RandomDouble` *(especially after seeing its one use)*

This makes my life easier in #4973 (where I'd really just implement `[0..1)` & then add/multiply based on min/max)

Malkierian replied:
Are you sure you didn't mean to have RandomDouble match Random_Float, so you could use RandomDouble for a range wider than 0...1?

serprex commented on soh/soh/Enhancements/randomizer/3drando/random.cpp:
```diff
@@ -25,10 +24,22 @@ uint32_t Random(int min, int max) {
+    return distribution(generator);
+}
+
 // Returns a random floating point number in [0.0, 1.0]
```
```suggestion
// Returns a random floating point number in [0.0, 1.0)
```

Malkierian replied:
This one doesn't make sense, the range opens with a [ in the comment.

Malkierian replied:
Ohhh, wait, that's actual range notation, isn't it? [] would be inclusive, and () denotes an end of a range that isn't inclusive?

serprex replied:
Yes

serprex commented on soh/soh/Enhancements/randomizer/3drando/random.cpp:
```diff
@@ -25,10 +24,22 @@ uint32_t Random(int min, int max) {
     return distribution(generator);
 }
 
+// Returns a random float in range [min, max-1]
+float Random_Float(float min, float max) {
+    Random_InitSeed();
+    boost::random::uniform_real_distribution<float> distribution(min, max - 1);
```
Do you really want this range? Seems like you'd want `[min, max)` instead of `[min, max-1)`

Also note this will never return `max-1` which I'm guessing you expected here

https://en.cppreference.com/w/cpp/numeric/random/uniform_real_distribution

Malkierian replied:
Idk, I was just matching the original Random's implementation.

Malkierian commented:
> Which compiler are you looking to reduce warnings from?

Oh, right, this is primarily on MSVC, since I'm on Windows, but I would be surprised if some of them weren't also resolving things on other compilers, like the implicit conversion warnings.

Malkierian committed:
Remove `Random_Float`, and apply `RandomDouble` where it was being used instead.
Add proper range notation to `RandomDouble` to denote the lack of inclusivity for the max part of the range.

briaguya0 reviewed (commented):
non-blocking: since all of these casts are in cpp i'd prefer to have them as [`static_cast`](https://en.cppreference.com/w/cpp/language/static_cast)s instead of c-style casts

Malkierian committed:
Convert c-style casts to static_cast.

Malkierian commented:
Done. Will merge when builds finish.
